171293217 wi fight - Leed descripcion

mmur07 opened the pull request:
Terminado el modo de juego Wi-fight

**Nuevas clases**
- WiFightGameMode: Nuevo gamemode que genera una entidad Router. La puntuación de cada jugador se guarda en un vector de doubles playerProgress_.
El router está compuesto por una fixture que solo rebota con las paredes y otra sensor que detecta jugadores.

-RouterLogic: Cuando un jugador entra en el área de su fixture sensor, lo registra en un vector. Cuando el jugador sale del área, lo expulsa del vector. En cada ciclo del update suma puntos a cada jugador en el área dependiendo de lo lejos que estén y del número de jugadores en el área. Un jugador gana cuando llega a 100 puntos.

**Cambios en otras clases**
- Collider: Creada la capa uninteractableObject y arreglado el horror de las colisiones entre capas. Ahora ya no hay un método CreateFixture. Hay un método CreateRectangularFixture y otro CreateCircularFixture. Necesitaba el segundo para crear el sensor del router.

- GameMode: En vez de tener un vector roundResults de Entidades jugador, ahora solo se registra 1 único jugador en la variable Winner. Para adaptar estos cambios he modificado la clase StocksGameMode.

- CollisionHandler: Comprueba colisiones entre el router y los jugadores.

mmur07 committed:
*screams in colliders*

title

mmur07 committed:
Casi terminado. Leer descripción

Ya van tanto el tema colisiones del router como tema lógica del mismo router.

Cuando detecta a un jugador, el router guarda su collider junto a su ID en un vector.
En cada update, el router le dice al modo de juego que le sume x puntos a el jugador en cuestión.
Cuando el jugador sale del área del router, se elimina del vector del router y por tanto deja de ganar puntos.

Todos los datos que necesitas están en el struct PlayerInfo del Router (Collider del jugador para su posición y su ID), y en un puntero al collider del router en la misma clase RouterLogic.

eljorgemo committed:
puntos cambiantes según la distancia del jugador al router

eljorgemo committed:
Merge branch 'master' into 171293217-Wi-Fight

eljorgemo committed:
Algunos arreglillos pero sigue sin compilar

eljorgemo committed:
Collision layers adjustment

eljorgemo committed:
it wORKS

mmur07 committed:
fixes finales

Pj6595 reviewed (changes requested):
No puedo probarlo porque en ningún sitio se dice al jugador su progreso y no se sabe si está funcionando bien. Comentadlo por consola aunque sea plz

mmur07 committed:
arreglado para que muestre couts

Pj6595 reviewed (approved):
Va bien, me gusta el sistema de ganar más puntos cuanto más te acercas.

EstebenR reviewed (commented):
Sale constantemente que el jugador ha ganado aunque no llegue a cien puntos?

mmur07 commented:
> 
> 
> Sale constantemente que el jugador ha ganado aunque no llegue a cien puntos?

No se si es a lo que te refieres pero cuando el primer jugador llega a 100 el booleando roundFinished se pasa a true. 
Ahora mismo no hacemos nada con ese booleano y se siguen sumando puntos todo el rato y por eso sale en consola que "ganan" todos los jugadores. En un futuro cuando ese booleano se vuelva true te sacará del modo de juego así que no problemo. Igual debería parar el update del router si roundFinished es true tho.

AdriAlvarez00 reviewed (approved):
Muy guay, aunque quitaría los debug antes de mergearlo

EstebenR committed:
Arreglado orden de Collider

EstebenR committed:
Añadido constants.json y made debugging easier

Ahora constants.json aparece en el proyecto para encontrarlo mas facil y cada vez que alguien gana puntos sale cuantos lleva en vez de cuantos ha ganado
